Microsoft.Toolkit.Uwp.UI/Extensions/Mouse/Mouse.cs

```diff
@@ -0,0 +1,115 @@
+        /// </summary>
+        public static readonly DependencyProperty CursorProperty =
+            DependencyProperty.RegisterAttached("Cursor", typeof(CoreCursorType), typeof(Mouse), new PropertyMetadata(CoreCursorType.Arrow, CursorChanged));
+
+        /// <summary>
+        /// Set the target <see cref="CoreCursorType"/>.
+        /// </summary>
+        /// <param name="element">Object where the selector cursor type should be shown.</param>
+        /// <param name="value">Target cursor type value.</param>
+        public static void SetCursor(FrameworkElement element, CoreCursorType value)
+        {
+            element.SetValue(CursorProperty, value);
+        }
+
+        /// <summary>
+        /// Get the current <see cref="CoreCursorType"/>.
+        /// </summary>
+        /// <param name="element">Object where the selector cursor type should be shown.</param>
+        /// <returns>Cursor type set on target element.</returns>
+        public static CoreCursorType GetCursor(FrameworkElement element)
+        {
+            return (CoreCursorType) element.GetValue(CursorProperty);
+        }
+
+        private static void CursorChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var element = d as FrameworkElement;
+            if (element == null)
+            {
+                throw new NullReferenceException(nameof(element));
+            }
+
+            var value = (CoreCursorType) e.NewValue;
+
+            // lock ensures CoreCursor creation and event handlers attachment/detachment is atomic
+            lock (s_cursorLock)
+            {
+                if (!s_cursors.ContainsKey(value))
+                {
+                    s_cursors[value] = new CoreCursor(value, 1);
+                }
+
+                // make sure event handlers are not attached twice to element
+                element.PointerEntered -= Element_PointerEntered;
+                element.PointerEntered += Element_PointerEntered;
+                element.PointerExited -= Element_PointerExited;
+                element.PointerExited += Element_PointerExited;
+                element.Unloaded -= ElementOnUnloaded;
+                element.Unloaded += ElementOnUnloaded;
+            }
+        }
+
+        private static void Element_PointerEntered(object sender, PointerRoutedEventArgs e)
+        {
+            CoreCursorType cursor = GetCursor((FrameworkElement) sender);
+            Window.Current.CoreWindow.PointerCursor = s_cursors[cursor];
+        }
+
+        private static void Element_PointerExited(object sender, PointerRoutedEventArgs e)
+        {
+            // when exiting change the cursor to the target Mouse.Cursor value of the new element
+            CoreCursor cursor;
+            if (e.OriginalSource is FrameworkElement newElement)
+            {
+                cursor = s_cursors[GetCursor(newElement)];
+            }
+            else
+            {
+                cursor = s_defaultCursor;
```

Comment:
Why do you need the else ? if should never reach this point, right ?
you should cast directly because this event is registered if the element is FrameworkElement.

Reply:
e.OriginalSource is the new object where the cursor is now, when it left my FrameworkElement with Mouse.Cursor set, but as far as I know all UI elements visible on the page must inherit from FrameworkElement, so this cast might not be necessary. I should probably remove this unnecessary if.

Reply:
Just realized this IF is required in case e.OriginalSource is null. This could happen when moving out touch pointer.